### PR TITLE
[SL-2391] Feature: panel pushing content library

### PR DESCRIPTION
**Pull request by freixedelo**
Panel pushing content to left removing backdrop. Adds dragging bat to resize in larger screens. 
on mobile panel still goes over the content.  
Added close button and copy icons to connect modal.

* [x] Feature

#### Jira

https://skylarkplatform.atlassian.net/browse/SL-2391

### CODE REVIEW

File: src/components/betaReleaseAuthModal/betaReleaseAuthModal.component.tsx [hunk 1]
```diff
@@ -102,29 +103,36 @@ export const AddAuthTokenModal = ({
...
               tabIndex={-1}
               className={clsx(
+                "pr-9",
                 debouncedUri && "border-2 outline-none",
                 requestLoading && "border-warning",
                 !requestLoading &&
                   debouncedUri &&
                   (invalidUri ? "border-error" : "border-success"),
               )}
+              copyIcon
```
Comment (james-wallis) [inline src/components/betaReleaseAuthModal/betaReleaseAuthModal.component.tsx]: withCopy

File: src/components/contentLibrary/contentLibrary.component.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+import { Panel } from "src/components/panel";
+
+export const ContentLibrary = () => {
+  const [activePanelObject, setActivePanelObject] = useState<{
+    objectType: string;
+    uid: string;
+  } | null>(null);
+
+  const [windowSize, setWindowSize] = useState(0);
+  const objectListingWidth = useMotionValue<number | undefined>(undefined); // from localstorage get saved sized
```
Comment (james-wallis) [inline src/components/contentLibrary/contentLibrary.component.tsx:15]: Remove this comment

File: src/components/contentLibrary/contentLibrary.component.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    updateWindowSize();
+
+    window.addEventListener("resize", updateWindowSize);
+    return () => {
+      window.removeEventListener("resize", updateWindowSize);
+    };
+  }, [activePanelObject, objectListingWidth]);
+
+  const handleDrag = React.useCallback(
+    (event: unknown, info: { delta: { x: number } }) => {
```
Comment (james-wallis) [inline src/components/contentLibrary/contentLibrary.component.tsx:32]: would prefer `event` tobe something other than `unknown`

File: src/components/contentLibrary/contentLibrary.component.tsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        objectListingWidth.get() || objectListingRef?.current?.offsetWidth || 0;
+      const newWidth = width + info.delta.x;
+      if (newWidth > 425 && newWidth < windowSize - 375) {
+        objectListingWidth.set(newWidth);
+      }
+    },
+    [objectListingWidth, windowSize],
+  );
+
+  const objecListingSize = clsx(
```
Comment (james-wallis) [inline src/components/contentLibrary/contentLibrary.component.tsx:43]: Move this directly into the `classname` on line 57 so all the class is together

File: src/components/objectListing/objectListing.component.tsx [hunk 5]
```diff
@@ -149,7 +162,7 @@ const createColumns = (
           editRowEnabled={opts.withObjectEdit}
           inEditMode={table.options.meta?.rowInEditMode === row.id}
           onEditClick={() => table.options.meta?.onEditClick(row.id)}
-          onInfoClick={() => setPanelInfo({ objectType, uid })}
+          onInfoClick={() => setPanelInfo?.({ objectType, uid })}
```
Comment (james-wallis) [inline src/components/objectListing/objectListing.component.tsx:165]: I think we should only add the `actionColumn` to the `orderedColumnArray` when `setPanelInfo` is truthy , like we have for `opts.withObjectSelect`

**Review by james-wallis — CHANGES_REQUESTED** [review]

File: src/components/contentLibrary/contentLibrary.component.tsx (hunk 4 above)
Comment (freixedelo) [reply src/components/contentLibrary/contentLibrary.component.tsx:43]: good spot

**Review by james-wallis — APPROVED** [review]
Looks good!

**Pull request merged**